lib/openstax/exchange/client/client.rb

```diff
@@ -0,0 +1,67 @@
+module OpenStax
+  module Exchange
+    module Client
+
+      @use_real_client = true
+
+      def self.configure
+        yield configuration
+      end
+
+      def self.configuration
+        @configuration ||= ClientConfiguration.new
+      end
+
+      def self.reset!
+        @client = nil
+      end
+
+      def self.use_real_client
+        @use_real_client = true
+      end
+
+      def self.use_fake_client
+        @use_real_client = false
+      end
+
+      def self.use_real_client?
+        !!@use_real_client
+      end
+
+      def self.create_identifier
+        begin
+          client.create_identifier
+        rescue StandardError => error
+          raise ClientError.new("create_identifier failure", error)
+        end
+      end
+
+      def self.create_multiple_choice(*args)
+        begin
+          client.create_multiple_choice(*args)
+        rescue StandardError => error
```

Comment:
Since there are going to be like 30 of these calls can we do:

```
def self.create_multiple_choice(arg1, arg2)
  delegate_to_client(:create_multiple_choice, arg1, arg2)
end

def delegate_to_client(method, *args)
  begin
    client.send(method, args)
  rescue StandardError => error
    raise ClientError.new("#{method.to_s} failure", error)
  end
end
```

eliminates copying the boilerplate begin rescue end all over the place.